### ISSUE TITLE
Revit could not complete the availability command

### ISSUE THREAD

**Issue by chauhuh** [eirannejad/pyRevit#345]
"pyRevitCorepyRevitpyRevittoolsReloadAvailb"

Revit encounter a Could not load file or assembly 'file:///C:User...\pyRevit_2018_f506ca8bbfc67605_pyRevitCore.dll' or one of its dependecies. The process cannot access the file because it is being used by another process. (Exception from HRESULT: 0X80070020)

I have uninstalled and reinstalled with the same error. Initial install worked fine, wrote some scripts, then this occured and I'm stuck. This error will repeat for all tools upon clicking on the pyRevit Tab.

Thanks.

**Comment by eirannejad** [comment]
Hmm. Any other errors on startup? Close Revit and clean the `%appdata%/pyRevit/<revit-version>/` cache files and start Revit again and see if the error still occurs.

**Comment by chauhuh** [comment]
So this is odd. I came in this morning with no restart and no changes that I know of and now everything works. I will keep your suggestion in mind if it crops up again.

There were no other error messages other than the one mentioned. It just occurred about 20 times, I believe once for each tool.

Thanks.

**Comment by eirannejad** [comment]
Yes Revit removes the call for the availability method if it fails for whatever reason the first time. So you get a call per tool and if they fail on any tool, you won't see any errors after that because Revit is not calling the availability method anymore for that tool.

**Comment by chauhuh** [comment]
This problem occurred again and I checked the folder that you recommended. The odd thing that I noticed was that the only DLL in the folder was the PyRevitBaseClasses.dll. Comparing this (2018) folder to the other folders it appears that a couple of DLL were missing.

I then reinstalled and tested 2018, and 2019. Again, only the PyRevitBaseClasses.dll was present along with some cache files. Seems like the PyRevitCore.dll, PyRevitTools.dll is not getting generated.

I also tested 2017 and was able to get it to run. First run only PyRevitBaseClasses.dll and it errored out. Second run, it generated the files and was able to run properly.

I have attached my first run 2018 log for your reference.

[pyRevit_2018_3672.log](https://github.com/eirannejad/pyRevit/files/2138320/pyRevit_2018_3672.log)

**Comment by chauhuh** [comment]
Ok, so reinstalled one more time and this gets odd. I installed as current user only vs all users for the previous installs but I the results were the same.

However, this time I was monitoring the install folder for 2018 when I started up revit. I watched it create the missing DLLs and then promptly delete them. No idea what is going on.

![pyrevit](https://user-images.githubusercontent.com/25804189/41931235-5b27dd28-7932-11e8-9823-1887fc3a85cb.gif)
